pkg/cmd/pr/checks/checks.go

```diff
@@ -81,7 +81,7 @@ func checksRun(opts *ChecksOptions) error {
 
 	rollup := pr.Commits.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
 	if len(rollup) == 0 {
-		return nil
+		return fmt.Errorf("no checks are set up for this repository")
```

Comment:
I like that the command is exiting with a nonzero status now! However, I do not think that this error message will always be accurate:

1. The branch for the PR might not report any checks if the branch was just created, so before any Checks nor Statuses got registered by external services yet. Re-running the command a few seconds later could show that there are now checks registered as “pending”.

2. The branch for the PR might not have associated checks because the base branch for the PR is not configured to run checks. However, a PR based on a different branch might have checks.

So, I think a better error message should be something more vague, like `no checks reported on the 'foo' branch`.

Should we also exit with nonzero status in the above `len(pr.Commits.Nodes)` condition where the PR does not appear to have any commits?

Reply:
I agree with you about changing the message according to the branch, that make more sense.
And for the empty commits case, maybe `no commit found on the pull request` would just be ok